### ISSUE TITLE
Search Error

### ISSUE THREAD

**Issue by omar-asfuor** [ekondur/ef-datatable.js#131]
System.InvalidOperationException: 'The LINQ expression 'DbSet<Audit>()
    .Where(a => a.TableName.IndexOf(
        value: "AmlElement", 
        comparisonType: OrdinalIgnoreCase) != -1)' could not be translated. Additional information: Translation of method 'string.IndexOf' failed. If this method can be mapped to your custom function

**Comment by ekondur** [comment]
Hi @omar-asfuor 
What is your EF version?

**Comment by ItzFender** [comment]
@ekondur,

I also get this error using Entity Framework Core 7.0.15. I believe it has something to do with the navigational properties on the entity even though they're not being used on the Datatable. I'm not 100% sure about that, just an assumption. In the past I had to create a SQL view of the entity and use that instead to get by the error. It would be nice to be able to use the entity itself.

**Comment by ItzFender** [comment]
I did some more testing on this and found it has something to do with the changes in DatatableJS.Data 3.7.0. I rolled the version back to 3.4.1 in my project and it started working again. Does this have something to do with the case sensitive searching that was added in 3.7.0?

Here's the full exception that's being thrown:

```
The LINQ expression 'DbSet<EntityTable>()
    .Where(v => v.EntityColumn.IndexOf(
        value: "test", 
        comparisonType: OrdinalIgnoreCase) != -1)' could not be translated. Additional information: Translation of method 'string.IndexOf' failed. If this method can be mapped to your custom function, see https://go.microsoft.com/fwlink/?linkid=2132413 for more information. Either rewrite the query in a form that can be translated, or switch to client evaluation explicitly by inserting a call to 'AsEnumerable', 'AsAsyncEnumerable', 'ToList', or 'ToListAsync'. See https://go.microsoft.com/fwlink/?linkid=2101038 for more information.
```

This seems to be the culprit in ExpressionBuilder.cs:
```
case Operand.Contains:
                    if (filter.CaseSensitive)
                    {
                        return Expression.Call(member, containsMethod, constant);
                    }
                    else
                    {
                        return Expression.NotEqual(Expression.Call(member, indexOfMethod, constant, ignoreCase), Expression.Constant(-1, typeof(int))); 
                    }
```

Before 3.7.0 it used to default to the CaseSensitve line which is fine because all searches in SQL server is already case insensitive. Right now for a column search the filter.CaseSensitive is always false and there's no way to set it to true as seen below, it's not set on the FilterDef so it's going to be false. DatatableExtensions.cs:

```
foreach (var item in request.columns.Where(a => a.searchable == true && !string.IsNullOrEmpty(a.search.value)))
            {
                ParameterExpression param = Expression.Parameter(typeof(T), "t");
                MemberExpression member = Expression.Property(param, item.name);
                var operand = member.Type == typeof(string) ? Operand.Contains : Operand.Equal;
                listExp.Add(new FilterDef { Operand = operand, Field = item.name, Value = item.search.value, Operator = Operator.And });
            }
```

My assumption is that this is not working for anyone that is doing server side processing and is using SQL Server or any other database that defaults to case insensitive searching.

**Comment by ekondur** [comment]
Hi @ItzFender 
I will fix it as soon as possible, thank you for your explanation

**Comment by ekondur** [comment]
Fixed with version 3.8.0, thank you for your contributions @ItzFender @omar-asfuor
